python/taichi/ad/_ad.py

```diff
@@ -92,7 +95,7 @@ def grad(self):
         self.gradient_evaluated = True
 
 
-def clear_all_gradients():
+def clear_all_gradients(gradient_type=SNodeGradType.ADJOINT):
```

Comment:
my gut feeling from this api name is that by default it clears all types of gradients :P

Reply:
The api name is indeed somehow misleading. I think the initial meaning of the api name might be clearing all the adjoint fields (because there is no other grad type at that time).